internal/dataplane/parser/translate_httproute.go

```diff
@@ -147,9 +142,6 @@ func generateKongRoutesFromHTTPRouteRule(httproute *gatewayv1alpha2.HTTPRoute, r
 			if match.Path != nil {
 				// determine the path match values
 				r.Route.Paths = []*string{match.Path.Value}
```

Comment:
So it seems the crux of this is that kong will already support regex paths by just looking to see if the path is a regex or not rather than needing to be explicitly flagged to expect a regex. That works for now, but I think we should make a comment here pointing this out:

```suggestion
				// the match path value can either be a literal path OR a regex: kong will
				// parse paths to determine if they are regex as opposed to needing to be
				// flagged.
				r.Route.Paths = []*string{match.Path.Value}
```